src/app/ReadHandler.cpp

```diff
@@ -153,6 +171,55 @@ CHIP_ERROR ReadHandler::ProcessReadRequest(System::PacketBufferHandle aPayload)
     return err;
 }
 
+CHIP_ERROR ReadHandler::ProcessAttributePathList(AttributePathList::Parser & aAttributePathListParser)
+{
+    CHIP_ERROR err                = CHIP_NO_ERROR;
+    ClusterInfo * LastClusterInfo = nullptr;
+    TLV::TLVReader reader;
+    aAttributePathListParser.GetReader(&reader);
+
+    while (CHIP_NO_ERROR == (err = reader.Next()))
+    {
+        VerifyOrExit(TLV::AnonymousTag == reader.GetTag(), err = CHIP_ERROR_INVALID_TLV_TAG);
+        VerifyOrExit(TLV::kTLVType_List == reader.GetType(), err = CHIP_ERROR_WRONG_TLV_TYPE);
+        ClusterInfo * clusterInfo = nullptr;
+        AttributePathParams attributePathParams;
+        AttributePath::Parser path;
+        err = path.Init(reader);
+        SuccessOrExit(err);
+        err = path.GetNodeId(&(attributePathParams.mNodeId));
+        SuccessOrExit(err);
+        err = path.GetEndpointId(&(attributePathParams.mEndpointId));
+        SuccessOrExit(err);
+        err = path.GetClusterId(&(attributePathParams.mClusterId));
+        SuccessOrExit(err);
+        err = path.GetFieldId(&(attributePathParams.mFieldId));
+        SuccessOrExit(err);
+        err = InteractionModelEngine::GetInstance()->PushFront(clusterInfo, attributePathParams);
```

Comment:
why not `PushFront(mpClusterInfoList, attributeParams)` ?

The is null check should already be part of the push front.
When I asked for 'push front' method I assumed this was the usage. Do you  care about the ordering of items to match the incoming list?

Reply:
not care about ordering, agree, just update that part, thanks!